lib/hyperion_test/fake_server/types.rb

```diff
@@ -0,0 +1,7 @@
+class Hyperion
+  class FakeServer
+    MimicRoute = ImmutableStruct.new(:method, :path)
+    Rule = ImmutableStruct.new(:mimic_route, :headers, :handler, :rest_route)
+    Request = ImmutableStruct.new(:body)
```

Comment:
`Struct` and `ImmutableStruct` are pretty handy. They allow you to define simple data structure types in a single line, automatically define attribute accessors and a constructor, are convertable to hashes, you can enumerate their attributes, ...